cmd/syncthing/support_bundle.go

```diff
@@ -0,0 +1,82 @@
+package main
+
+import (
+	"archive/zip"
+	"io"
+	"log"
+	"os"
+)
+
+// ZipFiles compresses one or many files into a single zip archive file
+func ZipFiles(filename *os.File, files []string) error {
```

Comment:
That is, if this took a slice of `struct { Name string, Data io.Reader }` then the Data can be a `bytes.Buffer` and the rest of the zip code is roughly the same, minus open/close stuff.

Reply:
Also, I think you could zip to a io.Writer (http.ResponseWriter) avoiding creating a files on disk.